qfieldsync/core/cloud_api.py

```diff
@@ -114,13 +122,69 @@ def from_reply(reply: QNetworkReply) -> Optional[CloudException]:
     return CloudException(reply, Exception(message))
 
 
+class CloudAuthMethod(Enum):
+    NONE = 0
+    CREDENTIALS = 1
+    SSO = 2
+
+
+def build_oauth2_auth_config_from_cloud_method(
```

Comment:
I do not follow `from_cloud_method` suffix here? Is it needed, or can we rename to something that is easier to follow?